packages/contracts-bedrock/contracts/L1/SystemConfig.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import {
+    OwnableUpgradeable
+} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import { Semver } from "../universal/Semver.sol";
+
+/**
+ * @title SystemConfig
+ * @notice This contract is used to update L2 configuration via L1
+ */
+contract SystemConfig is OwnableUpgradeable, Semver {
+    uint256 public constant VERSION = 0;
+
+    uint256 public overhead;
+    uint256 public scalar;
+    bytes32 public batcherHash;
+
+    event ConfigUpdate(
+        uint256 indexed version,
+        UpdateType indexed updateType,
+        bytes data
+    );
+
+    enum UpdateType {
+        BATCHER,
+        GAS_CONFIG
+    }
+
+    constructor(address _owner, uint256 _overhead, uint256 _scalar, bytes32 _batcherHash) Semver(0, 0, 1) {
+        overhead = _overhead;
+        scalar = _scalar;
+        initialize(_owner, _batcherHash);
```

Comment:
We should be setting `overhead` and `scalar` in the `initialize` function. Perhaps via the functions so that events are emitted instead of setting them directly. If we don't go through `initialize`, then the values will not be set in the proxy

Reply:
I'm following the pattern of l2 ouput oracle here where protocol params are set in the constructor, and addresses/permissions are set in initialize. Is the output oracle set up wrong?

Reply:
Exactly, immutables must be set in the constructor and not the `initialize` function

Reply:
Updated